### ISSUE TITLE
RPC return value for 0

### ISSUE THREAD

**Issue by ethers** [ethereum/go-ethereum#578]
The return value for 0 should be 0x

In addition to the 2 cases below, please review and fix any other APIs.  (The 2 cases below are eth_getBalance and eth_getTransactionCount returning 0x0 and 0 respectively.  Note, with eth_getTransactionCount the spec says the return value should be hex string https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactioncount )

```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["0xfd4ed114ef85d350d6d40ed3f6dc23743f8f99c4", "latest"],"id":1}' http://localhost:8080
{
	"id": 1,
	"jsonrpc": "2.0",
	"result": "0x0"
}%                                  
```

```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionCount","params":["0x9dc2299a76b68b7ffa9e3ba0fd8cd7646d21d400","pending"],"id":1}' http://localhost:8080
{
	"id": 1,
	"jsonrpc": "2.0",
	"result": 0
}%
```

**Comment by ethers** [comment]
Update per https://github.com/ethereum/cpp-ethereum/issues/1433 is 0 should be 0x0

**Comment by tgerring** [comment]
Fix merged in 779a51c0479ae5715d73db6ffb4edbef92d21ff3
